### ISSUE TITLE
Adding a ruby 2.2.3 fails

### ISSUE THREAD

**Issue by wendorf** [pivotal-sprout/sprout-chruby#2]
If I have ruby 2.2.3 in my list of rubies to install, when I `bundle exec soloist run_recipe sprout-chruby::rubies`, I receive the following error:

```
Generating RI format into /Users/pivotal/src/ruby-2.2.3/.ext/rdoc...
Apple LLVM version 7.0.0 (clang-700.0.72)
Target: x86_64-apple-darwin14.5.0
Thread model: posix

No newer files.
/Users/pivotal/src/ruby-2.2.3/lib/rubygems/installer.rb:645:in `verify_gem_home': You don't have write permissions for the /Library/Ruby/Gems/2.0.0 directory. (Gem::FilePermissionError)
    from /Users/pivotal/src/ruby-2.2.3/lib/rubygems/installer.rb:765:in `pre_install_checks'
    from /Users/pivotal/src/ruby-2.2.3/lib/rubygems/installer.rb:231:in `install'
    from /Users/pivotal/src/ruby-2.2.3/lib/rubygems/resolver/specification.rb:96:in `install'
    from /Users/pivotal/src/ruby-2.2.3/lib/rubygems/request_set.rb:257:in `block in install_into'
    from /Users/pivotal/src/ruby-2.2.3/lib/rubygems/request_set.rb:249:in `each'
    from /Users/pivotal/src/ruby-2.2.3/lib/rubygems/request_set.rb:249:in `install_into'
    from /Users/pivotal/src/ruby-2.2.3/lib/rubygems/request_set.rb:141:in `install'
    from /Users/pivotal/src/ruby-2.2.3/lib/rubygems/dependency_installer.rb:394:in `install'
    from /Users/pivotal/src/ruby-2.2.3/lib/rubygems.rb:558:in `install'
    from ./tool/rbinstall.rb:722:in `block (2 levels) in <main>'
    from ./tool/rbinstall.rb:721:in `each'
    from ./tool/rbinstall.rb:721:in `block in <main>'
    from ./tool/rbinstall.rb:757:in `call'
    from ./tool/rbinstall.rb:757:in `block in <main>'
    from ./tool/rbinstall.rb:754:in `each'
    from ./tool/rbinstall.rb:754:in `<main>'
make: *** [do-install-all] Error 1
!!! Installation of ruby 2.2.3 failed!
---- End output of ruby-install --cleanup --rubies-dir /Users/pivotal/.rubies ruby 2.2.3 ----
Ran ruby-install --cleanup --rubies-dir /Users/pivotal/.rubies ruby 2.2.3 returned 255
[2015-09-28T10:07:00-07:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

I believe there's some ruby/gem/bundler pollution happening with trying to install ruby 2.2.3 from a system ruby process. If I run `ruby-install --cleanup --rubies-dir /Users/pivotal/.rubies ruby 2.2.3` (the command that sprout-chruby::rubies actually runs) myself from a terminal, it succeeds.

**Comment by wendorf** [comment]
Did a little more testing. It looks like this failure happens on all 2.2.x rubies, and does not happen on 2.1.x rubies. I've had success on 2.1.5, 2.1.6, and 2.1.7. I've had failures on 2.2.3 and 2.2.0.

**Comment by wendorf** [comment]
The issue is due to [this commit](https://github.com/ruby/ruby/commit/a7f03a4659d34a459f915a41f260a1b9299ef219#diff-e2bd4eb9bbb8b25862825c636e55bd6d) to MRI Ruby that passes along the GEM_HOME environment variable to the Rubygems installer. Since we're running `ruby-install` from within system Ruby, GEM_HOME is set to the system Ruby home. If we clear that environment variable before executing ruby-install, we should be good.

PR incoming.

**Comment by wendorf** [comment]
Closed with #4
